Builds on Linux using cmake.

darktjm opened the pull request:
Found this repository when looking for C++ examples for panda3d.  Only 2, but better than none.   Here's a minor change to get it to build on Linux (for me).

darktjm committed:
Builds on Linux using cmake.
Also, use self-relative for nested local includes instead of path from parent

darktjm commented:
Guess I should've provided a top-level CMakeLists.txt as well.  I'm not going to add it to the pull request; instead, I leave it here:

>    cmake_minimum_required(VERSION 3.21.0)
>
>    project(c-p3d-samples LANGUAGES CXX)
>
>    subdirs(snake ball-in-maze)

IsakTheHacker commented:
Thank you very much for this! As you can probably tell, this repository hasn't been updated in a very long time as I haven't used Panda3D that much lately. I wish you the best trying to learn Panda3D as I remember the C++ documentation being quite bad. If you ever manage to translate a Python sample to C++, feel free to add it to this collection!

darktjm commented:
> If you ever manage to translate a Python sample to C++, feel free to add it to this collection!

I have converted most of the official samples, as well as some other code to C++ in my "fork". Since I changed the coding style significantly (including the one that you provided), and even removed "snake", it is probably not worth pulling into your repository.  The last commit that retained some of your coding style was added as a separate pull request (which added drain animations and on-screen help text).

I also found a more complete conversion (albeit old enough that even the C++ needs to change to be compatible with C++-17, the default used by gcc) at https://github.com/drivird/drunken-octo-robot which I probably would've looked at first, if it had actually showed up in my web searches.  That repository actually implements some things I didn't, so it's actually in some respects a better reference than mine.  Oh well.  I probably won't be wasting any more time on Panda3D.
